src/incubator/Dialog/useDialogCloseButton.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {PropsWithChildren, useMemo} from 'react';
+import {BorderRadiuses, Colors} from '../../style';
+import {DialogCloseButtonProps, DialogContentProps} from './types';
+import {StyleSheet} from 'react-native';
+import Assets from '../../assets';
+import DialogHeader from './DialogHeader';
+import Icon from '../../components/icon';
+import Text from '../../components/text';
+import TouchableOpacity from '../../components/touchableOpacity';
+import View from '../../components/view';
+
+interface InternalDialogCloseButtonProps extends DialogCloseButtonProps {
+  close: () => void;
+}
+
+const useDialogCloseButton = (props: InternalDialogCloseButtonProps) => {
+  const {
+    showClose,
+    close,
+    closeProps,
+    containerStyle: propsContainerStyle,
+    containerProps: propsContainerProps
+  } = props;
+
+  const DialogCloseButton = useMemo(() => {
+    if (!showClose) {
+      return null;
+    }
+
+    return (
+      <View left centerV pointerEvents={'box-none'}>
+        <TouchableOpacity paddingB-s2 row onPress={close}>
+          <Icon source={Assets.icons.xMedium} tintColor={Colors.white} {...closeProps?.iconProps}/>
+          <Text recorderTag={'unmask'} text70BO white {...closeProps?.textProps}>
+            {closeProps?.closeText || 'Close'}
```

Comment:
This will need to be configured in private to pass an i18n string

Reply:
Yes, that's why I added the props (text + text style + icon)